### ISSUE TITLE
[feature] tsconfig.json comment support

### ISSUE THREAD

**Issue by williamlark** [moonrepo/moon#1122]
**Is your feature request related to a problem? Please describe.**

When running tasks on a project that has a`tsconfig.json` with comments, the tasks fails. Seems it's related to the moon linting, but tsconfigs aren't strict JSON, and some tools auto-generate tsconfigs with comments. `tsc`'s default tsconfig template even uses comments.

**Describe the solution you'd like**

Would be great to modify the linting task in-built to moon to allow using comments in project tsconfig files.

**Comment by milesj** [comment]
@williamlark Comments in JSON should work. Can you include more info, like logs? The tsconfig that fails?

**Comment by williamlark** [comment]
Seem like `//` comments work fine, but `/* ... */` comments do not. Is this expected?

```json
/** This doesn't work **/
{
  "compilerOptions": {
    "noEmit": true
  }
}
```

```shell
Error: json::parse_file

  × Failed to parse JSON file ~/Documents/.../apps/frontend/tsconfig.json.
  ╰─▶ expected value at line 1 column 1
```

**Comment by milesj** [comment]
@williamlark It looks like the lib only supports `/* */` and not docblock syntax `/** **/` 

https://crates.io/crates/json_comments

**Comment by milesj** [comment]
I created an issue for it: https://github.com/tmccombs/json-comments-rs/issues/12

**Comment by milesj** [comment]
Ok fixed in 1.16.1
